### PR TITLE
Fixing principal

### PR DESCRIPTION
**Pull request by jorge-cr-13**

### CODE REVIEW

File: generic/lightsail_deployment/variables.tf [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  type        = string
+}
+
+variable "private_registry_access" {
+  description = "Describes a request to configure an Amazon Lightsail container service to access private container image repositories"
+  type        = bool
+  default     = false
+}
+
+variable "ecr_role" {
```
Comment (Robert-Steiner) [inline generic/lightsail_deployment/variables.tf:12]: it doesn't seem to be used

File: generic/lightsail_deployment/main.tf [hunk 2]
```diff
@@ -0,0 +1,29 @@
+resource "aws_lightsail_container_service_deployment_version" "this" {
+  count = var.private_registry_access && var.ecr_role != null ? 1 : var.private_registry_access ? 0 : 1
```
Comment (Robert-Steiner) [inline generic/lightsail_deployment/main.tf:2]: why do we only create a deployment when we have private_registry_access?

File: generic/lightsail_deployment/variables.tf (hunk 1 above)
Comment (jorge-cr-13) [reply generic/lightsail_deployment/variables.tf:12]: It is used for the count statement `count = var.private_registry_access && var.ecr_role != null ? 1 : var.private_registry_access ? 0 : 1`. 

This is the only way I found to make this resource dependent on the role.

File: generic/lightsail_deployment/main.tf (hunk 2 above)
Comment (jorge-cr-13) [reply generic/lightsail_deployment/main.tf:2]: This works if the `private_registry_access` is set to true, then create the deployment only if the `ecr_role` exists. If `private_registry_access` is false, then create the deployment.

File: generic/lightsail_deployment/variables.tf (hunk 1 above)
Comment (Robert-Steiner) [reply generic/lightsail_deployment/variables.tf:12]: did `dependencies` in terragrunt not work? we used that in the past for the es index and pg database https://github.com/xaynetwork/xayn_infrastructure/blob/ba3fe8e7a3365ada4f313f454b284950d2ee0506/b2b_personalization/infra/_env/tenant/users_api.hcl#L63

Comment (jorge-cr-13) [reply generic/lightsail_deployment/variables.tf]: Great! I was searching for something like that. I'll use this instead! :)

**Pull request merged**